src/network/mcpe/ProcessSkinTask.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\network\mcpe;
+
+use pocketmine\entity\InvalidSkinException;
+use pocketmine\entity\Skin;
+use pocketmine\network\mcpe\convert\TypeConverter;
+use pocketmine\network\mcpe\protocol\types\skin\SkinData;
+use pocketmine\scheduler\AsyncTask;
+use pocketmine\thread\NonThreadSafeValue;
+
+class ProcessSkinTask extends AsyncTask{
+	private const KEY_ON_COMPLETION = "onCompletion";
+
+	private ?string $error = "Unknown";
+	/** @var NonThreadSafeValue<SkinData> */
+	private NonThreadSafeValue $skinData;
+
+	/**
+	 * @param \Closure(?Skin $skin,?string $error) : void $callback
+	 */
+	public function __construct(
+		SkinData $skinData,
+		\Closure $callback,
+	){
+		$this->skinData = new NonThreadSafeValue($skinData);
+		$this->storeLocal(self::KEY_ON_COMPLETION, $callback);
+	}
+
+	public function onRun() : void{
+		try{
+			$skin = TypeConverter::getInstance()->getSkinAdapter()->fromSkinData($this->skinData->deserialize());
+			$this->setResult($skin);
+			$this->error = null;
+		}catch(\InvalidArgumentException|InvalidSkinException $e){
+			$this->error = $e->getMessage();
+		}
+	}
+
+	public function onCompletion() : void{
+		$result = $this->getResult();
+		if(!($result instanceof Skin)){
+			$result = null;
```

Comment:
Not sure I understand the purpose of this check. It's never going to be anything but `Skin|null`. A `@var` would be fine.

(Yes, I know I usually say don't use `@var`, but in this specific case it's the norm because `AsyncTask` is poorly designed.)